libyul/Object.cpp

```diff
@@ -65,9 +65,18 @@ set<YulString> Object::dataNames() const
 {
 	set<YulString> names;
 	names.insert(name);
-	for (auto const& subObject: subIndexByName)
-		names.insert(subObject.first);
-	// The empty name is not valid
+
+	for (std::shared_ptr<ObjectNode> const& subObjectNode: subObjects)
+	{
+		auto const* subObject = dynamic_cast<Object*>(subObjectNode.get());
+		if (subObject)
+		{
+			names.insert(subObject->name);
+			for (YulString const& subObjectName: subObject->dataNames())
+				names.insert(YulString{name.str() + "." + subObjectName.str()});
+		}
+	}
+
```

Comment:
I added these changes to recursively go into subobjects even though they are not accessible... I can change this to only add direct subobjects.

Reply:
You mean non accessible via the Assembly mechanism? Then please restrict to accessible objects only.